app/src/main/java/ai/elimu/appstore/synchronization/AppListArrayAdapter.java

```diff
@@ -66,12 +94,179 @@ public View getView(int position, @Nullable View convertView, @NonNull ViewGroup
                 public void onClick(View view) {
                     Timber.i("buttonDownload onClick");
 
+                    Timber.i("Downloading " + application.getPackageName() + " version " + application.getVersionCode());
+
+                    viewHolder.buttonDownload.setVisibility(View.GONE);
+                    viewHolder.progressBarDownloadProgress.setVisibility(View.VISIBLE);
+                    viewHolder.textViewDownloadProgress.setVisibility(View.VISIBLE);
+
                     // Initiate APK download
-                    // TODO
+                    ApplicationVersionGson applicationVersionGson = new ApplicationVersionGson(); // TODO: use application.getApplicationVersions().get(0);
+                    // TODO: replace with application.getApplicationVersions().get(0)
+                    ApplicationGson applicationGson = new ApplicationGson();
+                    applicationGson.setPackageName(application.getPackageName());
+                    applicationVersionGson.setApplication(applicationGson);
+                    applicationVersionGson.setFileUrl("/apk/" + application.getPackageName() + "-" + application.getVersionCode() + ".apk");
+                    applicationVersionGson.setVersionCode(application.getVersionCode());
+                    applicationVersionGson.setFileSizeInKb(69437);
```

Comment:
What is this hardcoded file size of 69437?

Reply:
It was for testing the progressbar with a specific APK file, but this code has now been replaced with dynamic values.